### ISSUE TITLE
Issue with Centrosome/propagate.pyx Package

### ISSUE THREAD

**Issue by cliandouble08** [CellProfiler/centrosome#118]
Hi, while installing centrosome package for CellProfiler. I encountered a persistent error message for the Centrosome/_propagate.pyx package as below: 

`s.py", line 1301, in cythonize_one
          raise CompileError(None, pyx_file)
      Cython.Compiler.Errors.CompileError: centrosome/_propagate.pyx
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed`

Is this a package issue or my environment set-up? 

Thanks.

**Comment by ebnertom** [comment]
I've experienced the same when installing centrosome.
environment:
* Ubuntu 20.04.6 LTS

```
#10 15.36   Preparing metadata (pyproject.toml): finished with status 'error'
#10 15.37   error: subprocess-exited-with-error
#10 15.37   
#10 15.37   × Preparing metadata (pyproject.toml) did not run successfully.
#10 15.37   │ exit code: 1
#10 15.37   ╰─> [60 lines of output]
#10 15.37       <string>:6: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
#10 15.37       /tmp/pip-build-env-5oyk2q5s/normal/lib/python3.9/site-packages/Cython/Compiler/Main.py:381: FutureWarning: Cython directive 'language_level' not set, using '3str' for now (Py3). This has changed from earlier releases! File: /tmp/pip-install-teaa56f0/centrosome_726d19d76c8e4d568f788ae5a3beb156/centrosome/_convex_hull.pyx
#10 15.37         tree = Parsing.p_module(s, pxd, full_module_name)
#10 15.37       /tmp/pip-build-env-5oyk2q5s/normal/lib/python3.9/site-packages/Cython/Compiler/Main.py:381: FutureWarning: Cython directive 'language_level' not set, using '3str' for now (Py3). This has changed from earlier releases! File: /tmp/pip-install-teaa56f0/centrosome_726d19d76c8e4d568f788ae5a3beb156/centrosome/_cpmorphology2.pyx
#10 15.37         tree = Parsing.p_module(s, pxd, full_module_name)
#10 15.37       /tmp/pip-build-env-5oyk2q5s/normal/lib/python3.9/site-packages/Cython/Compiler/Main.py:381: FutureWarning: Cython directive 'language_level' not set, using '3str' for now (Py3). This has changed from earlier releases! File: /tmp/pip-install-teaa56f0/centrosome_726d19d76c8e4d568f788ae5a3beb156/centrosome/_fastemd.pyx
#10 15.37         tree = Parsing.p_module(s, pxd, full_module_name)
#10 15.37       /tmp/pip-build-env-5oyk2q5s/normal/lib/python3.9/site-packages/Cython/Compiler/Main.py:381: FutureWarning: Cython directive 'language_level' not set, using '3str' for now (Py3). This has changed from earlier releases! File: /tmp/pip-install-teaa56f0/centrosome_726d19d76c8e4d568f788ae5a3beb156/centrosome/_filter.pyx
#10 15.37         tree = Parsing.p_module(s, pxd, full_module_name)
#10 15.37       /tmp/pip-build-env-5oyk2q5s/normal/lib/python3.9/site-packages/Cython/Compiler/Main.py:381: FutureWarning: Cython directive 'language_level' not set, using '3str' for now (Py3). This has changed from earlier releases! File: /tmp/pip-install-teaa56f0/centrosome_726d19d76c8e4d568f788ae5a3beb156/centrosome/_lapjv.pyx
#10 15.37         tree = Parsing.p_module(s, pxd, full_module_name)
#10 15.37       warning: centrosome/_lapjv.pyx:281:12: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
#10 15.37       warning: centrosome/_lapjv.pyx:281:18: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
#10 15.37       /tmp/pip-build-env-5oyk2q5s/normal/lib/python3.9/site-packages/Cython/Compiler/Main.py:381: FutureWarning: Cython directive 'language_level' not set, using '3str' for now (Py3). This has changed from earlier releases! File: /tmp/pip-install-teaa56f0/centrosome_726d19d76c8e4d568f788ae5a3beb156/centrosome/_propagate.pyx
#10 15.37         tree = Parsing.p_module(s, pxd, full_module_name)
#10 15.37       
#10 15.37       Error compiling Cython file:
#10 15.37       ------------------------------------------------------------
#10 15.37       ...
#10 15.37             heap.ptrs[child][k] = new_elem[k]
#10 15.37         heap.items += 1
#10 15.37       
#10 15.37         # restore heap invariant, all parents <= children
#10 15.37         while child>0:
#10 15.37             parent = (child + 1) / 2 - 1 # __parent(i)
#10 15.37                                      ^
#10 15.37       ------------------------------------------------------------
#10 15.37       
#10 15.37       centrosome/heap.pxd:148:31: Cannot assign type 'double' to 'unsigned int'
#10 15.37       Compiling centrosome/_propagate.pyx because it changed.
#10 15.37       Compiling centrosome/_convex_hull.pyx because it changed.
#10 15.37       Compiling centrosome/_filter.pyx because it changed.
#10 15.37       Compiling centrosome/_lapjv.pyx because it changed.
#10 15.37       Compiling centrosome/_cpmorphology2.pyx because it changed.
#10 15.37       Compiling centrosome/_fastemd.pyx because it changed.
#10 15.37       [1/6] Cythonizing centrosome/_convex_hull.pyx
#10 15.37       [2/6] Cythonizing centrosome/_cpmorphology2.pyx
#10 15.37       [3/6] Cythonizing centrosome/_fastemd.pyx
#10 15.37       [4/6] Cythonizing centrosome/_filter.pyx
#10 15.37       [5/6] Cythonizing centrosome/_lapjv.pyx
#10 15.37       [6/6] Cythonizing centrosome/_propagate.pyx
#10 15.37       Traceback (most recent call last):
#10 15.37         File "/usr/local/lib/python3.9/dist-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
#10 15.37           main()
#10 15.37         File "/usr/local/lib/python3.9/dist-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
#10 15.37           json_out['return_val'] = hook(**hook_input['kwargs'])
#10 15.37         File "/usr/local/lib/python3.9/dist-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 149, in prepare_metadata_for_build_wheel
#10 15.37           return hook(metadata_directory, config_settings)
#10 15.37         File "/tmp/pip-build-env-5oyk2q5s/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 380, in prepare_metadata_for_build_wheel
#10 15.37           self.run_setup()
#10 15.37         File "/tmp/pip-build-env-5oyk2q5s/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 487, in run_setup
#10 15.37           super(_BuildMetaLegacyBackend,
#10 15.37         File "/tmp/pip-build-env-5oyk2q5s/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 338, in run_setup
#10 15.37           exec(code, locals())
#10 15.37         File "<string>", line 89, in <module>
#10 15.37         File "/tmp/pip-build-env-5oyk2q5s/normal/lib/python3.9/site-packages/Cython/Build/Dependencies.py", line 1134, in cythonize
#10 15.37           cythonize_one(*args)
#10 15.37         File "/tmp/pip-build-env-5oyk2q5s/normal/lib/python3.9/site-packages/Cython/Build/Dependencies.py", line 1301, in cythonize_one
#10 15.37           raise CompileError(None, pyx_file)
#10 15.37       Cython.Compiler.Errors.CompileError: centrosome/_propagate.pyx
#10 15.37       [end of output]
#10 15.37   
#10 15.37   note: This error originates from a subprocess, and is likely not a problem with pip.
#10 15.37 error: metadata-generation-failed
#10 15.37 
#10 15.37 × Encountered error while generating package metadata.
#10 15.37 ╰─> See above for output.
#10 15.37 
#10 15.37 note: This is an issue with the package mentioned above, not pip.
#10 15.37 hint: See above for details.
------

```

**Comment by gnodar01** [comment]
Resolved by #116
